lib/model/QubitDigitalObject.php

```diff
@@ -1577,14 +1631,24 @@ public static function getCount($mediaTypeId, $options = array())
     return BasePeer::doCount($criteria)->fetchColumn(0);
   }
 
+  /**
+   * Check whether the master is stored within AtoM
+   *
+   * @return boolean  whether or not master is stored
+   */
+  public function masterStored($usageId)
+  {
+    return QubitTerm::EXTERNAL_URI_ID != $usageId && QubitTerm::EXTERNAL_FILE_ID != $usageId;
```

Comment:
Should we check `QubitTerm::OFFLINE_ID` too?

Reply:
Maybe we could use this function in a few more of the checks you're modifying, including the ones with `QubitTerm::OFFLINE_ID` if we add it.

It's just a note as I'm not sure if the three should have the same behavior in all the cases.

Reply:
OFFLINE_ID is different in that no derivatives are created at all (only metadata, or a digital object stored elsewhere, is stored in the digital object). I've changed the name of the `masterStored` method to reflect this difference (changed to `derivativesGeneratedFromExternalMaster`).